R/read_from_dhis2-helpers.R

```diff
@@ -109,7 +116,9 @@ dhis2_subset_fields <- function(data, fields) {
 #' }
 #' @keywords internal
 #' @noRd
-dhis2_subset_records <- function(data, records, id_col_name) {
+dhis2_subset_records <- function(data,
+                                 records,
+                                 id_col_name = "dataElement") {
```

Comment:
Is this the same variable that is named `which` in other function?

Reply:
When subsetting a vector of records from a data frame, the name of the column that unique identifies the subjects is need. DHIS2 data are usually uniquely identified based on either their `dataElement` or `dataSets` or `OrganisationUnits` columns depending on whether you are fetching the `dataElements` or the `dataSets` or `organisationUnits`, etc.

However, the data that interest a user is generally obtained by combining these different entities. In this case the returned data will contains all these columns togethers with remaining columns.

But I am assuming for DHIS2 that users subset the entries based on the `dataElement` column. I must say that will not be always the case.

The `which` argument is used to specify which data the user is interested in. It is "by coincidence" if the attribute name (dataElement or dataSet) is also the name of the column that unique identifies the data from that attribute.
Hope that this make sens.